go-controller/pkg/ovn/egressip.go

```diff
@@ -2480,23 +1924,84 @@ func (e *egressIPController) createReroutePolicyOps(ops []ovsdb.Operation, podIP
-	gatewayRouterIP, err := e.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
-	if err != nil {
-		return nil, fmt.Errorf("unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %w", status.Node, isEgressIPv6, err)
+	var nextHopIP string
+	var err error
+	isLocalZoneNode, ok := e.localZoneNodes.Load(status.Node)
+	if !config.OVNKubernetesFeature.EnableInterconnect || (ok && isLocalZoneNode.(bool)) {
+		gatewayRouterIP, err := e.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
+		if err != nil {
+			return nil, fmt.Errorf("unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %w", status.Node, isEgressIPv6, err)
+		}
+		nextHopIP = gatewayRouterIP.String()
+	} else {
+		// fetch node annotation of the egress node
+		nextHopIP, err = e.getTransitSwitchIP(status.Node)
+		if apierrors.IsNotFound(err) {
+			// if the node is already gone, then don't count it as error as during deletion.
+			klog.Warningf("Unable to fetch transit switch IP for node: %s: err: %v", status.Node, err)
+		} else if err != nil {
+			return nil, fmt.Errorf("unable to fetch transit switch IP for node %s: %v", status.Node, err)
+		}
 	}
-
 	// Handle all pod IPs that match the egress IP address family
 	for _, podIPNet := range util.MatchAllIPNetFamily(isEgressIPv6, podIPNets) {
 		filterOption := fmt.Sprintf("%s.src == %s", ipFamilyName(isEgressIPv6), podIPNet.IP.String())
 		p := func(item *nbdb.LogicalRouterPolicy) bool {
 			return item.Match == filterOption && item.Priority == types.EgressIPReroutePriority && item.ExternalIDs["name"] == egressIPName
 		}
-		ops, err = libovsdbops.DeleteNextHopFromLogicalRouterPoliciesWithPredicateOps(e.nbClient, ops, types.OVNClusterRouter, p, gatewayRouterIP.String())
+		ops, err = libovsdbops.DeleteNextHopFromLogicalRouterPoliciesWithPredicateOps(e.nbClient, ops, types.OVNClusterRouter, p, nextHopIP)
 		if err != nil {
 			return nil, fmt.Errorf("error removing nexthop IP %s from egress ip %s policies on router %s: %v",
-				gatewayRouterIP, egressIPName, types.OVNClusterRouter, err)
+				nextHopIP, egressIPName, types.OVNClusterRouter, err)
+		}
+	}
+	return ops, nil
+}
+
+func (e *egressIPZoneController) createStaticRouteOps(ops []ovsdb.Operation, podIPNets []*net.IPNet, status egressipv1.EgressIPStatusItem, egressIPName string) ([]ovsdb.Operation, error) {
+	isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
+	gatewayRouterIP, err := e.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %w", status.Node, isEgressIPv6, err)
+	}
+	// Handle all pod IPs that match the egress IP address family
+	for _, podIPNet := range util.MatchAllIPNetFamily(isEgressIPv6, podIPNets) {
+		lrsr := nbdb.LogicalRouterStaticRoute{
+			IPPrefix: podIPNet.IP.String(),
+			Nexthop:  gatewayRouterIP.String(),
+			ExternalIDs: map[string]string{
+				"name": egressIPName,
```

Comment:
assuming this is a brand new static route we will be introducing in the codebase going fwd, could we add some additional external-ids here? Something explicit like **"owner": "hello-world-I-am-an-egress-ip-static-route"**  
but better than that, of course :) I ask because having something like that makes the syncStaticRoutes a lot safer to prune stale stuff. And also a todo placeholder here to ensure that sync is able to remove/fix these static routes!

Reply:
hmm do we need another externalID in addition to the egressIPName one? technically speaking that is the owner and we use that for deleting stuff like we do for other objects..

Reply:
My thinking is that tables like static routes are shared amongst different functionalities on ovnk8. Reconciling it may be a little tricky. But I do agree that managing more things can be a possibility of getting more bugs down the road.
Cost-wise, I think that another key:value is not a big deal. The secondary indexing functionality makes it super fast too. Let's see what others think about that and see if this is a good idea or a silly idea. ;)

Reply:
we should use unique ids with client side indexes like @jcaamano and @npinaeva added for other resource types. ACL example: https://github.com/ovn-org/ovn-kubernetes/pull/3521

Then you dont need to do predicate searches on source ip, next hop, etc, we just use the key and do fast lookup. Not sure this is in scope for this PR though as it would probably require also adding db indexes for all other static routes that we add. Also, see my earlier comment about why do we even use static routes.

Reply:
I agree we should used indexes if there is perf gain. That will need to be a new epic since eIP code is so complex, don't want to couple with IC rn.